inference/worker/work.py

```diff
@@ -40,9 +41,20 @@ def _prepare_message(message: inference.MessageRead) -> str:
     # Construct prompt
     messages = [_prepare_message(message) for message in work_request.thread.messages]
 
-    # Prepend system prompt if it was specified in work parameters
-    if work_request.parameters.system_prompt:
-        pre_prompt = V2_SYSTEM_PREFIX + work_request.parameters.system_prompt + eos_token
+    # Prepend system prompt and custom_instructions if it was specified in work parameters
+    if (
+        work_request.parameters.system_prompt
+        or work_request.parameters.user_profile
+        or work_request.parameters.user_response_instructions
+    ):
+        work_params = work_request.parameters
+        if work_params.system_prompt or work_params.user_profile or work_params.user_response_instructions:
+            pre_prompt = V2_SYSTEM_PREFIX + (work_params.system_prompt or "")
```

Comment:
This can be simplified a bit (currently the same `if` condition is repeated)
```suggestion
    work_params = work_request.parameters
    if work_params.system_prompt or work_params.user_profile or work_params.user_response_instructions:
        pre_prompt = V2_SYSTEM_PREFIX + (work_params.system_prompt or "")
```

Reply:
Yes, I just badly copy/pasted. I committed your change, Thanks!